### ISSUE TITLE
Virtual Columns with custom sql queries

### ISSUE THREAD

**Issue by dahyui** [propelorm/Propel2#974]
Hi,

I've written a custom sql query that returns product details and one extra field (min price).

How do I access that virtual column for the min price? When I look at the object, there are no virtual columns available?

Thanks,
Darryn

**Comment by SourceCode** [comment]
Without example code its very difficult to say.  Are you using withColumn or another method that will include and build that virtual column?  I believe you have to use specific methods to add your column declaration in order for a virtual column and its corresponding methods to be created.

The way you access virtual columns is like any other column if you included the a column such as 

withColumn("a.min_price", "MinPrice");

After you execute that query you can access MinPrice via $myData->getMinPrice();

Hope this helps!

**Comment by marcj** [comment]
Please provide us example code so we can see if you have made an mistake or something else is wrong.

**Comment by dahyui** [comment]
Hi, apologies for the delay, I've been away.

So what I did was I wrote my sql statement out and executed the statement (I've used a basic sql query for example purposes): 
$sql = "SELECT p.*, pp.minPrice
	    FROM products p 
	    INNER JOIN productprices pp ON pp.product_id = p.product_id"

$stmt->execute();
$formatter = new ObjectFormatter();
$formatter->setClass('\Products'); //full qualified class name
$searchquery_product = $formatter->format($con->getDataFetcher($stmt));

After getting the data, I loop through each object and each object does not have a virtual column.

Your help would be greatly appreciated.

Thanks
